chain/network/src/test_loop.rs

```diff
@@ -0,0 +1,16 @@
+use near_primitives::types::AccountId;
+
+/// A multi-instance test using the TestLoop framework can support routing
+/// lookup for network messages, as long as the Data type contains AccountId.
+/// This trait is just a helper for looking up the index.
+pub trait SupportsRoutingLookup {
+    fn index_for_account(&self, account: &AccountId) -> usize;
+}
+
+impl<InnerData: AsRef<AccountId>> SupportsRoutingLookup for Vec<InnerData> {
+    fn index_for_account(&self, account: &AccountId) -> usize {
+        self.iter()
+            .position(|data| data.as_ref() == account)
+            .expect(&format!("Account not found: {}", account))
+    }
```

Comment:
Are you sure you want this interface to be placed here? In golang the rule is to define the interface in the package that needs it https://github.com/golang/go/wiki/CodeReviewComments#interfaces , and I guess same would apply here, especially since you define only a blanket implementation for an abstract type.

Reply:
Hmm, I expect this to be used for any multi-instance test for nearcore, because routing a network message will need an account ID lookup for most cases. So it needs to belong to somewhere lower in the stack. Will move this if it doesn't end up being appropriate, and hopefully future network stack refactorings will make this all different and easier.